test/test_utils/testHelpers.ts

```diff
@@ -0,0 +1,140 @@
+import { ethers, ContractTransactionResponse, AddressLike } from "ethers";
+import { HardhatEthersHelpers } from "hardhat/types/runtime";
+import {
+    MathStateChannelManagerProxy,
+    MathStateMachine,
+    StateChannelUtilLibrary
+} from "@typechain-types";
+
+import { JoinChannelStruct } from "@typechain-types/contracts/V1/DataTypes";
+import Clock from "@/Clock";
+import P2pEventHooks from "@/P2pEventHooks";
+
+export const createJoinChannelTestObject = (
+    address: AddressLike,
+    channelId?: string
+): JoinChannelStruct => {
+    let currentTime = 0;
+    try {
+        currentTime = Clock.getTimeInSeconds();
+    } catch (e) {
+        currentTime = Math.floor(Date.now() / 1000);
+    }
+    let jc: JoinChannelStruct = {
+        participant: address,
+        channelId: channelId
+            ? ethers.keccak256(
+                  ethers.AbiCoder.defaultAbiCoder().encode(
+                      ["string"],
+                      [channelId]
+                  )
+              )
+            : ethers.keccak256("0x2371"),
+        amount: 500,
+        deadlineTimestamp: currentTime + 120, // 2 minutes from now
+        data: "0x00"
+    };
+    return jc;
+};
+
+export const getCurrentBlockTime = async (
+    provider: ethers.Provider
+): Promise<number> => {
+    const block = await provider.getBlock("latest");
+    return block!.timestamp;
+};
+export const getCurrentTimeSeconds = (): number => {
+    return Math.floor(Date.now() / 1000);
+};
+
+export async function deployLibraryTestContract(
```

Comment:
I think also we should specify deployment on chain and offchain in naming

Reply:
Sure - 99% of deployments in tests are on-chain - currently only the specific state machine (Math, Tic-Tac-Toe) is deployed 'off-chain' and we don't deploy it in tests, we just generate a deployment transaction and pass it to the SDK which deploys it internally.

In general the solidity tests are tested normally with hardhat and other Ethereum tooling without knowing about the SDK. Think of it as unit tests and even the stuff that runs locally (off-chain) can be tested 'on-chain' normally, since it's the same EVM. 

For off-chain specifically we can use 'deployLocal'